src/test/java/edu/unc/lib/boxc/migration/cdm/services/RedirectMappingIndexServiceTest.java

```diff
@@ -0,0 +1,195 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.boxc.migration.cdm.services;
+
+import edu.unc.lib.boxc.migration.cdm.exceptions.InvalidProjectStateException;
+import edu.unc.lib.boxc.migration.cdm.model.MigrationProject;
+import edu.unc.lib.boxc.migration.cdm.options.SipGenerationOptions;
+import edu.unc.lib.boxc.migration.cdm.test.SipServiceHelper;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author snluong
+ */
+public class RedirectMappingIndexServiceTest {
+    private static final String PROJECT_NAME = "proj";
+    private static final String USERNAME = "migr_user";
+    private static final String DEST_UUID = "7a33f5e6-f0ca-461c-8df0-c76c62198b17";
+
+    @Rule
+    public final TemporaryFolder tmpFolder = new TemporaryFolder();
+
+    private MigrationProject project;
+    private RedirectMappingIndexService indexService;
+    private SipService sipsService;
+    private SipServiceHelper testHelper;
+
+    @Before
+    public void setup() throws Exception {
+        tmpFolder.create();
+        project = MigrationProjectFactory.createMigrationProject(
+                tmpFolder.getRoot().toPath(), PROJECT_NAME, null, "user");
+        testHelper = new SipServiceHelper(project, tmpFolder.newFolder().toPath());
+        sipsService = testHelper.createSipsService();
+        indexService = new RedirectMappingIndexService(project);
+        indexService.setConnectionString("jdbc:sqlite:" + testHelper.getRedirectMappingIndexPath());
+
+        Connection conn = indexService.openDbConnection();
+        Statement statement = conn.createStatement();
+        statement.execute("drop table if exists redirect_mappings");
+        statement.execute("CREATE TABLE redirect_mappings (" +
+                "id INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                "cdm_collection_id varchar(64) NOT NULL, " +
+                "cdm_object_id varchar(64) NOT NULL, " +
+                "boxc_work_id varchar(64) NOT NULL, " +
+                "boxc_file_id varchar(64) DEFAULT NULL, " +
+                "created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP, " +
+                "UNIQUE (cdm_collection_id, cdm_object_id)" +
+                ")");
+        CdmIndexService.closeDbConnection(conn);
+    }
+
+    @Test
+    public void indexingDoesNotHappenIfCollectionIsNotSubmitted() {
+        try {
+            indexService.indexMapping();
```

Comment:
after this line, add `fail();` (its another junit assertion method), otherwise the test will pass even if no exception is thrown